documentation/modules/oauth/proc-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,632 @@
+ClusterManager of my-cluster Group has full access to topics on my-cluster
+----
++
+Dev Team A:: The Dev Team A realm role can write to topics that start with `x_` on any cluster. This combines a resource called `Topic:x_*`, `Describe` and `Write` scopes, and the `Dev Team A` policy. The `Dev Team A` policy matches all users that have a realm role called `Dev Team A`.
+Dev Team B:: Dev Team B realm role can read from topics that start with `x_` on any cluster. This combines `Topic:x_*`, `Group:x_*` resources, `Describe` and `Read` scopes, and the `Dev Team B` policy. The `Dev Team B` policy matches all users that have a realm role called `Dev Team B`. Matching users and clients have the ability to read from topics, and update the consumed offsets for topics and consumer groups that have names starting with `x_`.
+
+[id='proc-oauth-authorization-keycloak-example-deploy-kafka_{context}']
+== Deploying a Kafka cluster with Keycloak authorization
+
+Deploy a Kafka cluster configured to connect to the Keycloak server.
+Use the example `kafka-ephemeral-oauth-single-keycloak-authz.yaml` file to deploy the Kafka cluster as a `Kafka` custom resource.
+The example deploys a single-node Kafka cluster with `keycloak` authorization and `oauth` authentication.
+
+.Prerequisites
+
+* The Keycloak authorization server is deployed to your Kubernetes cluster and loaded with the example realm.
+* The Cluster Operator is deployed to your Kubernetes cluster.
+* The Strimzi `examples/security/keycloak-authorization/kafka-ephemeral-oauth-single-keycloak-authz.yaml` custom resource.
+
+.Procedure
+
+. Use the hostname of the Keycloak instance you deployed to prepare a truststore certificate for Kafka brokers to communicate with the Keycloak server.
++
+[source,shell,subs="+quotes"]
+----
+SSO_HOST=_SSO-HOSTNAME_
+SSO_HOST_PORT=$SSO_HOST:443
+STOREPASS=storepass
+
+echo "Q" | openssl s_client -showcerts -connect $SSO_HOST_PORT 2>/dev/null | awk ' /BEGIN CERTIFICATE/,/END CERTIFICATE/ { print $0 } ' > /tmp/sso.crt
+----
++
+The certificate is required as Kubernetes ingress is used to make a secure (HTTPS) connection.
+
+. Deploy the certificate to Kubernetes as a secret.
++
+[source,shell]
+----
+kubectl create secret generic oauth-server-cert --from-file=/tmp/sso.crt -n $NS
+----
+
+. Set the hostname as an environment variable
++
+[source,shell,subs="+quotes"]
+----
+SSO_HOST=_SSO-HOSTNAME_
+EXAMPLES_DIR=examples
+cat $EXAMPLES_DIR/security/keycloak-authorization/kafka-ephemeral-oauth-single-keycloak-authz.yaml | sed -E 's#\${SSO_HOST}'"#$SSO_HOST#" | kubectl create -n $NS -f -
+----
+
+. Create and deploy the example Kafka cluster.
++
+[source,shell]
+----
+kubectl apply -f examples/security/keycloak-authorization/kafka-ephemeral-oauth-single-keycloak-authz.yaml
+----
+
+[id='proc-oauth-authorization-keycloak-example-authentication_{context}']
+== Authenticating Keycloak access to Kafka brokers
```

Comment:
@mstruk Is this title accurate? Are teh steps split correctly in this procedure?